lib/queuePopulator/QueuePopulator.js

```diff
@@ -219,14 +219,21 @@ class QueuePopulator {
     _setupUpdatedReaders(done) {
         const newReaders = this.logReadersUpdate;
         this.logReadersUpdate = null;
-        async.each(newReaders, (logReader, cb) => logReader.setup(cb),
-                   err => {
-                       if (err) {
-                           return done(err);
-                       }
-                       this.logReaders = newReaders;
-                       return done();
-                   });
+        async.each(
+            newReaders,
+            (logReader, cb) => logReader.setup(cb),
+            err => {
+                if (err) {
+                    return done(err);
+                }
+                return async.each(
+                    this.logReaders, (logReader, cb) => logReader.close(cb),
```

Comment:
should the error returned from `logReader.close` be ignored (ex.. with `() => cb()`)?

Reply:
In theory, I think you're right, we should not prevent other log readers from being closed if one of them has an error. In practice though, I believe `async.each` will just trigger all of them in parallel so it should not make any difference, hence looks more of a code style issue to me.